low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -61,62 +62,129 @@ def __init__(self):
-        logger.debug(self.log.svc_log(
-            f"Server Type:{server_type}"))
+        logger.debug(self.log.svc_log(f"Server Type:{server_type}"))
         supported_types = ["physical"]
         if server_type not in supported_types:
             msg = f"Health provider is not supported for server type:{server_type}"
             logger.error(self.log.svc_log(msg))
-            raise ResourceMapError(
-                errno.EINVAL, msg)
+            raise ResourceMapError(errno.EINVAL, msg)
 
     def get_health_info(self, rpath):
         """
-        Get health information of fru in given resource id
+        Fetch health information for given rpath
 
-        rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
+        rpath: Resource path to fetch its health
+               Examples:
+                    node>compute[0]
+                    node>compute[0]>hw
+                    node>compute[0]>hw>disks
         """
         logger.info(self.log.svc_log(
             f"Get Health data for rpath:{rpath}"))
         info = {}
-        fru_found = False
+        resource_found = False
         nodes = rpath.strip().split(">")
         leaf_node, _ = self.get_node_details(nodes[-1])
+
+        # Fetch health information for all sub nodes
         if leaf_node == "compute":
-            for fru in self.server_map:
+            info = self.get_server_health_info()
+            resource_found = True
+        elif leaf_node in self.server_resources:
+            for resource, method in self.server_resources[leaf_node].items():
                 try:
-                    info.update({fru: self.server_map[fru]()})
-                except:
-                    # TODO: Log the exception
-                    info.update({fru: None})
-            info["last_updated"] = int(time.time())
-            fru_found = True
+                    info.update({resource: method()})
+                    resource_found = True
+                except Exception as err:
+                    logger.error(
+                        self.log.svc_log(f"{err.__class__.__name__}: {err}"))
+                    info = None
         else:
-            fru = None
             for node in nodes:
-                fru, _ = self.get_node_details(node)
-                if self.server_map.get(fru):
-                    fru_found = True
+                resource, _ = self.get_node_details(node)
+                for res_type in self.server_resources:
+                    method = self.server_resources[res_type].get(resource)
+                    if not method:
+                        logger.error(
+                            self.log.svc_log(
+                                f"No mapping function found for {res_type}"))
+                        continue
                     try:
-                        info = self.server_map[fru]()
-                    except:
-                        # TODO: Log the exception
+                        info = method()
```

Comment:
parent key not need for single FRU?

Reply:
Consumer will have the response under the hierarchy.
```
{
  "node": {
    "storage": [
      {
        "hw": {
          "controllers": [
                 {<response>}
            ]
         }
       }
      ]
   }
}
```